art/estimators/classification/pytorch.py

```diff
@@ -484,13 +484,18 @@ def fit_generator(self, generator: "DataGenerator", nb_epochs: int = 20, **kwarg
 
     def clone_for_refitting(self) -> "PyTorchClassifier":  # lgtm [py/inheritance/incorrect-overridden-signature]
         """
-        Create a copy of the classifier that can be refit from scratch. Will inherit same architecture, optimizer and
+        Create a copy of the classifier that can be refit from scratch. Will inherit same architecture, same type of optimizer and
         initialization as cloned model, but without weights.
 
         :return: new estimator
         """
         model = copy.deepcopy(self.model)
-        clone = type(self)(model, self._loss, self.input_shape, self.nb_classes, optimizer=self._optimizer)
+
+        # create a new optimizer that binds to cloned model's parameters
+        new_optimizer = type(self._optimizer)(model.parameters())
+        new_optimizer.load_state_dict(self._optimizer.state_dict())
```

Comment:
Hi, thanks for the review. This line is intended to inherit any user-provided parameters like learning rate, but I'll check if there is any unintended side effects and whether there is a more explicit way to inherit the params (params_group?) without loading the whole `state_dict`.
For the resetting, wondering whether `zero_grad()` used in the training already serves the purpose, will check it out too. It may take me some time to get back on this though, apologies in advance.

Reply:
I think param_groups is probably the right way to go here. There may be other elements in the state besides the gradients (e.g., adaptive learning rate, weight decay, etc.) which would not be reset by zero_grad().

Reply:
Hi @abigailgold, thanks, removed `load_state_dict` and instead create the new optimizer with `optimizer.defaults`, which includes all the parameters provided by the user, including learning rate and weight decay etc. I suspect the defaults stay unchanged during training, but the parameters in param_groups may change (e.g. for adaptive learning rate).